File: app/src/main/java/com/example/cstore/common/base/BaseViewModel.java
package com.example.cstore.common.base;

import androidx.lifecycle.ViewModel;

import com.example.cstore.control.LocalControl;
import com.google.gson.Gson;

public abstract class BaseViewModel extends ViewModel {
    /* **********************************************************************
     * Variable
     ********************************************************************** */
    public Gson gson = new Gson();
    public LocalControl local = LocalControl.getInstance();
}

File: app/src/main/java/com/example/cstore/common/base/BaseViewModel.kt
package com.example.cstore.common.base

import androidx.lifecycle.ViewModel
import com.example.broadcastation.common.logger.Logger
import com.google.gson.Gson

open class BaseViewModel : ViewModel() {
    /* **********************************************************************
     * Variable
     ********************************************************************** */
    protected val gson = Gson()
    protected val logger = Logger.instance
}